### DISCUSSION TITLE
Automatically confirm bluetooth passkey

### DISCUSSION THREAD

**Discussion by mn-xd** [bluez/bluer#64] [Q&A] [answered]
Hello,

I am trying to find a way to auto confirm Bluetooth passkey request. I have GATT server running on my computer and I want to connect to it with my phone, but I cannot connect because of `Authentication failed` and also sometimes because of this error: `Remote User Terminated Connection (0x13)` but I think passkey should fix it

When I open bluetoothctl and GATT server, everything works, but I need to say yes to accept passkey request. I did find something in documentation, but I don't know how to implement it in code.
I used code from examples(gatt_server_io).
Furthermore, I also edited Bluetooth config to have 7 number passkey enabled because default was 0.

**Comment by surban** [comment, marked as answer]
Register an agent and handle the appropriate requests.

See https://docs.rs/bluer/0.15.4/bluer/agent/struct.Agent.html

You can check `gattcat` in the repository for an example.

**Reply by mn-xd** [reply]
Thank you
